### ISSUE TITLE
ActiveStorage on Rails 7.0.1 forces vips

### ISSUE THREAD

**Issue by dvodvo** [rails/rails#44211]
### Steps to reproduce

Gemfile uncommented  gem 'image-processing'. 
Gemfile.lock subsequently has
    image_processing (1.12.1)
      mini_magick (>= 4.9.5, < 5)
      ruby-vips (>= 2.0.17, < 3)
and
    ruby-vips (2.1.4)
      ffi (~> 1.12)

which is contrary to [what is indicated here](https://guides.rubyonrails.org/active_storage_overview.html#requirements).   "libvips v8.6+ or ImageMagick for image analysis and transformations". **focus being on the 'or'**.

So it appears vips was forced in.

Thus when creating an image to be stored via an S3-compatible API an error is hit when invoking a variant. [paths popped for brevity sake]
```
 ActiveStorage::VariantRecord Load (0.5ms)  SELECT "active_storage_variant_records".* FROM "active_storage_variant_records" WHERE "active_storage_variant_records"."blob_id" = $1 AND "active_storage_variant_records"."variation_digest" = $2 LIMIT $3  [["blob_id", 6], ["variation_digest", "spWDgYmFptMnmwEsfcpfOPL1tZA="], ["LIMIT", 1]]
  S3 Storage (205.8ms) Downloaded file from key: l1ijfuheu8w8satmqpfgq2j9fv9z
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:51: warning: already initialized constant GLib::G_FREE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:51: warning: previous definition of G_FREE was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:59: warning: already initialized constant GLib::LOG_FLAG_RECURSION
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:59: warning: previous definition of LOG_FLAG_RECURSION was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:60: warning: already initialized constant GLib::LOG_FLAG_FATAL
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:60: warning: previous definition of LOG_FLAG_FATAL was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:63: warning: already initialized constant GLib::LOG_LEVEL_ERROR
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:63: warning: previous definition of LOG_LEVEL_ERROR was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:64: warning: already initialized constant GLib::LOG_LEVEL_CRITICAL
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:64: warning: previous definition of LOG_LEVEL_CRITICAL was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:65: warning: already initialized constant GLib::LOG_LEVEL_WARNING
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:65: warning: previous definition of LOG_LEVEL_WARNING was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:66: warning: already initialized constant GLib::LOG_LEVEL_MESSAGE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:66: warning: previous definition of LOG_LEVEL_MESSAGE was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:67: warning: already initialized constant GLib::LOG_LEVEL_INFO
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:67: warning: previous definition of LOG_LEVEL_INFO was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:68: warning: already initialized constant GLib::LOG_LEVEL_DEBUG
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:68: warning: previous definition of LOG_LEVEL_DEBUG was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:71: warning: already initialized constant GLib::GLIB_TO_SEVERITY
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:71: warning: previous definition of GLIB_TO_SEVERITY was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:86: warning: already initialized constant GLib::LOG_HANDLER
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:86: warning: previous definition of LOG_HANDLER was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:155: warning: already initialized constant GObject::GBOOL_TYPE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:155: warning: previous definition of GBOOL_TYPE was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:156: warning: already initialized constant GObject::GINT_TYPE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:156: warning: previous definition of GINT_TYPE was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:157: warning: already initialized constant GObject::GUINT64_TYPE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:157: warning: previous definition of GUINT64_TYPE was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:158: warning: already initialized constant GObject::GDOUBLE_TYPE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:158: warning: previous definition of GDOUBLE_TYPE was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:159: warning: already initialized constant GObject::GENUM_TYPE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:159: warning: previous definition of GENUM_TYPE was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:160: warning: already initialized constant GObject::GFLAGS_TYPE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:160: warning: previous definition of GFLAGS_TYPE was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:161: warning: already initialized constant GObject::GSTR_TYPE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:161: warning: previous definition of GSTR_TYPE was here
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:162: warning: already initialized constant GObject::GOBJECT_TYPE
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:162: warning: previous definition of GOBJECT_TYPE was here
Completed 500 Internal Server Error in 252ms (ActiveRecord: 2.8ms | Allocations: 31463)


#<Thread:0x0000000116677e58 .rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/live.rb:340 run> terminated with exception (report_on_exception is true):
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ffi-1.15.5/lib/ffi/library.rb:145:in `block in ffi_lib': Could not open library 'vips.42': dlopen(vips.42, 5): image not found. (LoadError)
Could not open library 'libvips.42.dylib': dlopen(libvips.42.dylib, 5): image not found
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ffi-1.15.5/lib/ffi/library.rb:99:in `map'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ffi-1.15.5/lib/ffi/library.rb:99:in `ffi_lib'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:573:in `<module:Vips>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:570:in `<main>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bootsnap-1.10.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:28:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bootsnap-1.10.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:28:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:35:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/image_processing-1.12.1/lib/image_processing/vips.rb:1:in `<main>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bootsnap-1.10.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:28:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bootsnap-1.10.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:28:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:35:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/transformers/image_processing_transformer.rb:26:in `const_get'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/transformers/image_processing_transformer.rb:26:in `processor'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/transformers/image_processing_transformer.rb:17:in `process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/transformers/transformer.rb:22:in `transform'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variation.rb:56:in `block in transform'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `block in instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variation.rb:55:in `transform'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variant_with_record.rb:35:in `block in transform_blob'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/downloader.rb:15:in `block in open'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/downloader.rb:24:in `open_tempfile'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/downloader.rb:12:in `open'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/service.rb:90:in `open'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/blob.rb:301:in `open'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variant_with_record.rb:34:in `transform_blob'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variant_with_record.rb:19:in `process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variant_with_record.rb:14:in `processed'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/controllers/active_storage/representations/base_controller.rb:14:in `set_representation'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:400:in `block in make_lambda'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:199:in `block (2 levels) in halting'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/abstract_controller/callbacks.rb:34:in `block (2 levels) in <module:Callbacks>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:200:in `block in halting'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:595:in `block in invoke_before'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:595:in `each'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:595:in `invoke_before'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:116:in `block in run_callbacks'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actiontext-7.0.1/lib/action_text/rendering.rb:20:in `with_renderer'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actiontext-7.0.1/lib/action_text/engine.rb:69:in `block (4 levels) in <class:Engine>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:127:in `instance_exec'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:127:in `block in run_callbacks'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:138:in `run_callbacks'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/abstract_controller/callbacks.rb:233:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/rescue.rb:22:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:67:in `block in process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `block in instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:66:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/params_wrapper.rb:259:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activerecord-7.0.1/lib/active_record/railties/controller_runtime.rb:27:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/abstract_controller/base.rb:151:in `process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionview-7.0.1/lib/action_view/rendering.rb:39:in `process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/live.rb:266:in `block (2 levels) in process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/concurrency/share_lock.rb:162:in `sharing'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/dependencies/interlock.rb:37:in `running'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/live.rb:258:in `block in process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/live.rb:343:in `block in new_controller_thread'
Exiting
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ffi-1.15.5/lib/ffi/library.rb:145:in `block in ffi_lib': Could not open library 'vips.42': dlopen(vips.42, 5): image not found. (LoadError)
Could not open library 'libvips.42.dylib': dlopen(libvips.42.dylib, 5): image not found
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ffi-1.15.5/lib/ffi/library.rb:99:in `map'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ffi-1.15.5/lib/ffi/library.rb:99:in `ffi_lib'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:573:in `<module:Vips>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/ruby-vips-2.1.4/lib/vips.rb:570:in `<main>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bootsnap-1.10.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:28:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bootsnap-1.10.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:28:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:35:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/image_processing-1.12.1/lib/image_processing/vips.rb:1:in `<main>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bootsnap-1.10.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:28:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/bootsnap-1.10.1/lib/bootsnap/load_path_cache/core_ext/kernel_require.rb:28:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/zeitwerk-2.5.3/lib/zeitwerk/kernel.rb:35:in `require'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/transformers/image_processing_transformer.rb:26:in `const_get'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/transformers/image_processing_transformer.rb:26:in `processor'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/transformers/image_processing_transformer.rb:17:in `process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/transformers/transformer.rb:22:in `transform'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variation.rb:56:in `block in transform'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `block in instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variation.rb:55:in `transform'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variant_with_record.rb:35:in `block in transform_blob'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/downloader.rb:15:in `block in open'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/downloader.rb:24:in `open_tempfile'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/downloader.rb:12:in `open'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/lib/active_storage/service.rb:90:in `open'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/blob.rb:301:in `open'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variant_with_record.rb:34:in `transform_blob'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variant_with_record.rb:19:in `process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/models/active_storage/variant_with_record.rb:14:in `processed'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activestorage-7.0.1/app/controllers/active_storage/representations/base_controller.rb:14:in `set_representation'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:400:in `block in make_lambda'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:199:in `block (2 levels) in halting'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/abstract_controller/callbacks.rb:34:in `block (2 levels) in <module:Callbacks>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:200:in `block in halting'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:595:in `block in invoke_before'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:595:in `each'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:595:in `invoke_before'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:116:in `block in run_callbacks'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actiontext-7.0.1/lib/action_text/rendering.rb:20:in `with_renderer'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actiontext-7.0.1/lib/action_text/engine.rb:69:in `block (4 levels) in <class:Engine>'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:127:in `instance_exec'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:127:in `block in run_callbacks'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/callbacks.rb:138:in `run_callbacks'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/abstract_controller/callbacks.rb:233:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/rescue.rb:22:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:67:in `block in process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `block in instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications/instrumenter.rb:24:in `instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/notifications.rb:206:in `instrument'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/instrumentation.rb:66:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/params_wrapper.rb:259:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activerecord-7.0.1/lib/active_record/railties/controller_runtime.rb:27:in `process_action'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/abstract_controller/base.rb:151:in `process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionview-7.0.1/lib/action_view/rendering.rb:39:in `process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/live.rb:266:in `block (2 levels) in process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/concurrency/share_lock.rb:162:in `sharing'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/activesupport-7.0.1/lib/active_support/dependencies/interlock.rb:37:in `running'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/live.rb:258:in `block in process'
.rbenv/versions/3.1.0/lib/ruby/gems/3.1.0/gems/actionpack-7.0.1/lib/action_controller/metal/live.rb:343:in `block in new_controller_thread'
```
Then the server quits,
when serving the tag
```
 <%= image_tag articlephoto.photo.variant(resize_to_fill: [500,500]) %>
```

imageMagick was already installed on ths local machine, but not vips & I had no intention of installing the latter.

In the meantime, the file did upload (DO Spaces) and is found with the image's blob key.  
`<%= image_tag articlephoto.photo %>`. 
renders fine.

My understanding is that the error occurs in the download process to the page (which effectively shows a broken icon) because the server is processing the image. As far as I know, there is no process for saving versions to the external service/CDN...

### Expected behavior

I expect either:
• based on the guide that vips not be invoked automatically.  
• be shown how to toggle between reliance on imageMagick vs vips

### Actual behavior

error above. 

Note, the first set of errors have `previous definition` and `already initialized`. statements which would be unexpected as well.

### System configuration
Rails 7.0.1

Ruby 3.1

**Comment by marcelolx** [comment]
What is the config you've set for `config.active_storage.variant_processor`? Are you on your config file calling `config.load_defaults 7.0`? If you have not set the `variant_processor` on your config file and are loading the default for rails 7 that's probably why it is using vips instead of image magick since Rails 7 defaults uses Vips instead ImageMagick.

**Comment by brenogazzola** [comment]
Rails does not force Vips. It is simply the default. You can change it to image magick in your environment files, as [indicated here](https://guides.rubyonrails.org/configuring.html#config-active-storage-variant-processor):

```ruby
config.active_storage.variant_processor = :mini_magick
```

**Comment by dvodvo** [comment]
That is useful.  However, the same paragrpah states ` The default is :mini_magick.`  
Having only one reference to activeStorage in `development.rb` 
```
config.active_storage.service = :digitalocean
```
this is still putting into question what the default is.

**Comment by marcelolx** [comment]
@dvodvo I think what is missing is update that doc, if you look at the upgrade guide you will see that the default did change to `:vips` https://guides.rubyonrails.org/upgrading_ruby_on_rails.html#active-storage-default-variant-processor-changed-to-vips

**Comment by dvodvo** [comment]
@marcelolx I agree.  I ran a clean install of 7.0.1, new application, to test things out.  I certainly was not consulting the 'upgrading' suggestions but focussed on the guide for active_storage.

**Comment by bjhess** [comment]
My experience following the guide to stand up a new Rails 7.0.1 app is that my app was looking for Vips while the guide says MiniMagick was the default processor. I had to add `config.active_storage.variant_processor = :mini_magick` to `application.rb` to get things working.

**Comment by clarkology56** [comment]
I tried adding   `config.active_storage.variant_processor = :mini_magick` to `application.rb` to get it to work. But, it made no difference. 

HOWEVER, When I added `  config.active_storage.variant_processor = :mini_magick` to `config/environments/test.rb` it fixed the issue. 

I assume there is some bug in setting the default, so I went ahead and added `  config.active_storage.variant_processor = :mini_magick` to all my environment files to be safe.
